internal/backends/postgresql/metadata/registry.go

```diff
@@ -73,10 +90,153 @@ func (r *Registry) Close() {
 	r.p.Close()
 }
 
-// getPool returns a pool of connections to PostgreSQL database
+// CheckAuthAndLoadMetadataIfEmpty returns a pool of connections to PostgreSQL database
 // for the username/password combination in the context using [conninfo].
 //
-// All methods should use that method to check authentication.
+// It loads metadata if it hasn't been loaded from the database yet.
+//
+// All methods should use this method to check authentication and load metadata.
+func (r *Registry) CheckAuthAndLoadMetadataIfEmpty(ctx context.Context) (*pgxpool.Pool, error) {
```

Comment:
There is no need for that method to be exported or be separated from `loadMetadata`